handlers/routeservice_test.go

```diff
@@ -620,4 +715,104 @@ var _ = Describe("Route Service Handler", func() {
 			Expect(nextCalled).To(BeFalse())
 		})
 	})
+	Context("allowlist wildcards resolve correctly", func() {
+
+		type args struct {
+			wildcardHost string
+		}
+
+		type testcase struct {
+			name    string
+			args    args
+			host    string
+			matched bool
+			err     bool
+		}
+		tests := []testcase{
+			{
+				name:    "Test invalid wildcard leading with 2 subdomains",
+				args:    args{"*.*.wildcard-a.com"},
+				host:    "first.authentication.wildcard-a.com",
+				matched: false,
+				err:     true,
+			},
+			{
+				name:    "Test wildcard in the wrong position",
+				args:    args{"first.*.wildcard-a.com"},
+				host:    "first.authentication.wildcard-a.com",
+				matched: false,
+				err:     true,
+			},
+			{
+				name:    "Test wildcard domain without path",
+				args:    args{"*.wildcard-a.com"},
+				host:    "authentication.wildcard-a.com",
+				matched: true,
+				err:     false,
+			},
+			{
+				name:    "Test wildcard domain is not a part of other domain",
+				args:    args{"*.wildcard-a.com"},
+				host:    "cola-wildcard-a.com",
+				matched: false,
+				err:     false,
+			},
+			{
+				name:    "Test wildcard for subdomain",
+				args:    args{"*.authentication.wildcard-a.com"},
+				host:    "first.authentication.wildcard-a.com",
+				matched: true,
+				err:     false,
+			},
+			{
+				name:    "Test wildcard for wrong domain on subdomain",
+				args:    args{"*.authentication.wildcard-a.com"},
+				host:    "first.authentication-wildcard-a.com",
+				matched: false,
+				err:     false,
+			},
+			{
+				name:    "Test fixed host name",
+				args:    args{"authentication.wildcard-a.com"},
+				host:    "authentication.wildcard-a.com",
+				matched: true,
+				err:     false,
+			},
+			{
+				name:    "Test wrong fixed host name",
+				args:    args{"authentication.wildcard-a.com"},
+				host:    "first.authentication.wildcard-a.com",
+				matched: false,
+				err:     false,
+			},
+			{
+				name:    "Test injecting a regex",
+				args:    args{"(.|\\w)+"},
+				host:    "first.authentication.wildcard-a.com",
+				matched: false,
+				err:     true,
+			},
```

Comment:
you could add a test for a broken allowlist entry, e.g.
```suggestion
			},
			{
				name:    "Test bad allowlist entry",
				args:    args{"*.ba d.cöm"},
				host:    "host.bad.com",
				matched: false,
				err:     true,
			},
```

Reply:
done.